docs: refactor (WIP) (#840)
aidenybai/million: website/pages/docs/index.mdx

import { Disclosures } from '../../components/home/faq';
import { Callout, Tab, Tabs } from 'nextra-theme-docs';
import Link from 'next/link';
import dynamic from 'next/dynamic';

export const Demo = dynamic(() =>
  import('../../components/demo').then((mod) => mod.Demo),
);

# Introduction

> **Million.js** is a fast and lightweight `<4kb` virtual DOM that makes React up to [70% faster](https://krausest.github.io/js-framework-benchmark/current.html). By using a fine-tuned, optimized virtual DOM, Million.js reduces the overhead of React; Imagine React components running at the speed of raw JavaScript.

Author: Hmm, thanks for this. Nice revision!

> **Million.js** can make your React applications faster with [Automatic Mode](/docs/automatic) enabled. Enabling auto-mode is simple - set up the Million compiler to handle optimizations automatically with just one parameter.

Author: Awesome! Sure thing, Let me do that.

> **Million.js** can make your React applications faster with [Automatic Mode](/docs/automatic) enabled. Enabling auto-mode is simple - set up the Million compiler to handle optimizations automatically with just one parameter.

Author: Done!

import { Disclosures } from '../../components/home/faq';
import { Callout, Tab, Tabs } from 'nextra-theme-docs';
import Link from 'next/link';
import dynamic from 'next/dynamic';

export const Demo = dynamic(() =>
  import('../../components/demo').then((mod) => mod.Demo),
);

# Introduction

> **Million.js** is an extremely fast and lightweight `<4kb` virtual DOM that makes React up to [70% faster](https://krausest.github.io/js-framework-benchmark/current.html). By using a fine-tuned, optimized virtual DOM, Million.js reduces the overhead of React.

Reviewer: ```suggestion
**Million.js** is an fast and lightweight (`<4kb`) virtual DOM that makes React up to [70% faster](https://krausest.github.io/js-framework-benchmark/current.html). By using a fine-tuned, optimized virtual DOM, Million.js reduces the overhead of React.
```

Author: Thanks for this Sarah, this is much appreciated!

> Imagine React components running at the speed of raw JavaScript.

Reviewer: This sentence feels orphaned. Might be worthwhile finding a better home for it or adding something after

Author: Hmm, thanks for this Sarah! This is much appreciated. I really appreciate it 🤔. I just tried something on it.

> **Million.js** makes your React applications faster automatically with auto-mode ([Automatic mode](/docs/automatic)) enabled. Enabling auto-mode is really easy to do, you just set up the Million compiler to handle optimizations
> automatically with literally one parameter. Check out the [Installation Guide](/docs/install) to get started right away.

Reviewer: ```suggestion
**Million.js** can make your React applications faster with [Automatic Mode](/docs/automatic) enabled. Enabling auto-mode is simple - set up the Million compiler to handle optimizations automatically with just one parameter. 

The [Installation Guide](/docs/install) can help you get started right away.
```

## Million.js vs. React

<Callout type="info">
  The following is a comprehensive demo using [key-based
  rendering](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key)
  to show how Million.js performance compares to React.
</Callout>

<br />

<Demo />

> ## Clarifications

Reviewer: ```suggestion
## Any questions?
```

> If you have questions or need clarity on things at this point, please feel free to ask them in the `questions` channel on the [Discord](https://discord.gg/7ZAGgTQVTE) or check out <Link href="/blog/million-beyond-speed">Million Beyond "Speed"</Link>

Reviewer: ```suggestion
If you have any questions or need support, please feel free to ask them in the `questions` channel on the [Discord](https://discord.gg/7ZAGgTQVTE) or check out <Link href="/blog/million-beyond-speed">Million Beyond "Speed"</Link>
```
